controllers/rollingupgrade_controller.go

```diff
@@ -795,6 +910,62 @@ Loop:
 	return nil
 }
 
+func (r *RollingUpgradeReconciler) UpdateInstanceEager(
+	ruObj *upgrademgrv1alpha1.RollingUpgrade,
+	nodeName,
+	targetInstanceID,
+	KubeCtlCall string,
+	ch chan error) {
+
+	// Set instance to standby.
+	err := r.SetStandby(ruObj, targetInstanceID)
+	if err != nil {
+		ch <- err
+		return
+	}
+
+	// Wait for new instance to be created
+	err = r.WaitForDesiredInstances(ruObj)
```

Comment:
Setting an instance into standby is best-effort. AWS might be at it's absolute limits for the account and may not be able to spin-up new instances. Even if the desired count was not reached, upgrade should proceed. WDYT?

Reply:
Yes
I think we need additional error handling right after `r.UpdateInstanceEager` or `r.DrainTerminate` are called.

The scenario where you wait for instances but they never show up is not easy to reproduce for testing with very short timeout.

Reply:
your assumption is correct from code perspective, if we time out, we move on and drain/kill, so this makes `eager` mode best-effort like you say.